src/mscorlib/shared/System/IO/BinaryWriter.cs

```diff
@@ -389,6 +389,16 @@ public unsafe virtual void Write(String value)
             }
         }
 
+        public virtual void Write(ReadOnlySpan<byte> span)
+        {
+            OutStream.Write(span);
```

Comment:
This unfortunately suffers a problem we've hit elsewhere: existing types that override methods on the type.  Because BinaryWriter isn't sealed and exposes virtual methods, someone could have derived from BinaryWriter and overridden all of the virtuals in order to provide custom behaviors.  By adding a new virtual, if new code calls that new virtual on a BinaryWriter they're passed, and that BinaryWriter is actually one of these derived implementations with all of the functionality overridden (except for the new members), we could end up providing the wrong behavior.  As such, I think we need to check whether this is actually a BinaryWriter or a derived type, and if it's the latter, delegate to one of the existing virtuals, e.g.
```C#
public virtual void Write(ReadOnlySpan<byte> span)
{
    if (GetType() == typeof(BinaryWriter))
    {
        OutStream.Write(span);
    }
    else
    {
        byte[] array = ArrayPool<byte>.Shared.Rent(span.Length);
        try
        {
            span.CopyTo(array);
            Write(array, 0, span.Length);
        }
        finally
        {
            ArrayPool<byte>.Shared.Return(array);
        }
    }
}
```